tests/testthat/test-sector_profile.R

```diff
@@ -127,3 +127,71 @@ test_that("at company level, two matched and one unmatched products yield `value
+  # * when: the corresponding value of `companies$tilt_sector` doesn't match any value of `scenarios$sector`
+  # * then: the result at product level has the value `NA` in `product$grouped_by`
+
+  # Based on this GoogleSheet row:
+  # https://docs.google.com/spreadsheets/d/16u9WNtVY-yDsq6kHANK3dyYGXTbNQ_Bn/edit#gid=156243064&range=A5:I5
+  # styler: off
+  companies <- tribble(
+    ~companies_id, ~clustered, ~activity_uuid_product_uuid, ~tilt_sector, ~tilt_subsector,       ~type,     ~sector,  ~subsector,
+              "a",        "b",                 "unmatched",  "unmatched",     "unmatched", "unmatched", "unmatched", "unmatched",
+  )
+  scenarios <- tribble(
+    ~sector,   ~subsector,  ~year, ~reductions, ~type, ~scenario,
+    "total",     "energy",   2050,         1.0, "ipr",       "a",
+  )
+  # styler: on
+
+  product <- sector_profile(companies, scenarios) |> unnest_product()
+
+  expect_true(is.na(product$grouped_by))
+})
+
+test_that("at product level, when `companies$tilt_sector` doesn't match `scenarios$sector` then `product$risk_category` and  `product$profile_ranking` are `NA`", {
+  # TODO: DRY with the test above
+  # styler: off
+  companies <- tribble(
+    ~companies_id, ~clustered, ~activity_uuid_product_uuid, ~tilt_sector, ~tilt_subsector,       ~type,     ~sector,  ~subsector,
+              "a",        "b",                 "unmatched",  "unmatched",     "unmatched", "unmatched", "unmatched", "unmatched",
+  )
+  scenarios <- tribble(
+    ~sector,   ~subsector,  ~year, ~reductions, ~type, ~scenario,
+    "total",     "energy",   2050,         1.0, "ipr",       "a",
+  )
+  # styler: on
+
+  product <- sector_profile(companies, scenarios) |> unnest_product()
+
+  # Tilman's expectations
+  # https://docs.google.com/spreadsheets/d/16u9WNtVY-yDsq6kHANK3dyYGXTbNQ_Bn/edit#gid=156243064&range=B19:D19
+  expect_true(is.na(product$risk_category))
+  expect_true(is.na(product$profile_ranking))
+})
+
+test_that("at product level, when `companies$sector`, `companies$subsector`, and `companies$type` match the corresponding columns in `scenarios`, then  `product$grouped_by` is '<type>_<scenario>_<year>' and `product$risk_category` and `product$profile_ranking` are not `NA`", {
+  # styler: off
+  companies <- tribble(
+    ~companies_id, ~clustered, ~activity_uuid_product_uuid, ~tilt_sector, ~tilt_subsector,       ~type,     ~sector,  ~subsector,
+              "a",        "a",                         "a",          "a",             "a",       "ipr",     "total",    "energy",
+  )
+  # matching type, sector and subsector
+  scenarios <- tribble(
+    ~sector,   ~subsector,  ~year, ~reductions, ~type, ~scenario,
+    "total",     "energy",   2050,         1.0, "ipr",       "a",
+  )
+  # styler: on
+
+  product <- sector_profile(companies, scenarios) |> unnest_product()
+
+  expect_equal(product$grouped_by, "ipr_a_2050")
+  expect_false(is.na(product$risk_category))
+  expect_false(is.na(product$profile_ranking))
+})
```

Comment:
This is the “happy path” at product level:

- Given: a value of `companies$clustered`, `companies$tilt_sector`, `companies$tilt_sector`, and `companies$type` ([example](https://docs.google.com/spreadsheets/d/16u9WNtVY-yDsq6kHANK3dyYGXTbNQ_Bn/edit#gid=156243064&range=B3:F3))
- Where: the value of `companies$type` matches `scenarios$type`, and the values of `companies$sector` and `companies$subsector` ([example](https://docs.google.com/spreadsheets/d/16u9WNtVY-yDsq6kHANK3dyYGXTbNQ_Bn/edit#gid=156243064&range=G3:H3)) matches [the values of `scenarios$sector` and `scenarios$subsector`](https://docs.google.com/spreadsheets/d/16u9WNtVY-yDsq6kHANK3dyYGXTbNQ_Bn/edit#gid=156243064&range=A11:B11))
- Then: the value of `product$grouped_by` is`‘<type>*<scenario>*<year>’`([example](https://docs.google.com/spreadsheets/d/16u9WNtVY-yDsq6kHANK3dyYGXTbNQ_Bn/edit#gid=156243064&range=B17)), and the value of `product$risk_cagegory` and `product$profile_ranking` are not `NA`.

<details/>
<summary/>
reprex
</summary>

``` r
withr::local_options(width = 500)
devtools::load_all()
#> ℹ Loading tiltIndicator

# styler: off
companies <- tribble(
  ~companies_id, ~clustered, ~activity_uuid_product_uuid, ~tilt_sector, ~tilt_subsector,       ~type,     ~sector,  ~subsector,
            "a",        "a",                         "a",          "a",             "a",       "ipr",     "total",    "energy",
)
# Matching type, sector and subsector
scenarios <- tribble(
  ~sector,   ~subsector,  ~year, ~reductions, ~type, ~scenario,
  "total",     "energy",   2050,         1.0, "ipr",       "a",
)
# styler: on

product <- sector_profile(companies, scenarios) |> unnest_product()
product
#> # A tibble: 1 × 11
#>   companies_id grouped_by risk_category profile_ranking clustered activity_uuid_product_uuid tilt_sector scenario  year type  tilt_subsector
#>   <chr>        <chr>      <chr>                   <dbl> <chr>     <chr>                      <chr>       <chr>    <dbl> <chr> <chr>         
#> 1 a            ipr_a_2050 high                        1 a         a                          a           a         2050 ipr   a
```

</details>